File: RealEstateSolution.Models/Contract.cs
using System.ComponentModel.DataAnnotations;

namespace RealEstateSolution.Models;

/// <summary>
/// 合同类型
/// </summary>
public enum ContractType
{
    /// <summary>
    /// 买卖合同
    /// </summary>
    Sale = 1,

    /// <summary>
    /// 租赁合同
    /// </summary>
    Rent = 2
}

/// <summary>
/// 合同状态
/// </summary>
public enum ContractStatus
{
    /// <summary>
    /// 草稿
    /// </summary>
    Draft = 1,

    /// <summary>
    /// 待签署
    /// </summary>
    Pending = 2,

    /// <summary>
    /// 已签署
    /// </summary>
    Signed = 3,

    /// <summary>
    /// 已完成
    /// </summary>
    Completed = 4,

    /// <summary>
    /// 已取消
    /// </summary>
    Cancelled = 5
}

/// <summary>
/// 合同模型
/// </summary>
public class Contract
{
    /// <summary>
    /// 合同ID
    /// </summary>
    [Key]
    public int Id { get; set; }

    /// <summary>
    /// 合同编号
    /// </summary>
    [Required]
    [MaxLength(50)]
    public string ContractNumber { get; set; } = string.Empty;

    /// <summary>
    /// 合同类型
    /// </summary>
    public ContractType Type { get; set; }

    /// <summary>
    /// 合同状态
    /// </summary>
    public ContractStatus Status { get; set; }

    /// <summary>
    /// 房产ID
    /// </summary>
    public int PropertyId { get; set; }

    /// <summary>
    /// 房产信息
    /// </summary>
    public Property? Property { get; set; }

    /// <summary>
    /// 客户ID
    /// </summary>
    public int ClientId { get; set; }

    /// <summary>
    /// 客户信息
    /// </summary>
    public Client? Client { get; set; }

    /// <summary>
    /// 合同金额
    /// </summary>
    public decimal Amount { get; set; }

    /// <summary>
    /// 合同开始日期
    /// </summary>
    public DateTime StartDate { get; set; }

    /// <summary>
    /// 合同结束日期
    /// </summary>
    public DateTime? EndDate { get; set; }

    /// <summary>
    /// 付款方式
    /// </summary>
    [MaxLength(200)]
    public string? PaymentMethod { get; set; }

    /// <summary>
    /// 合同条款
    /// </summary>
    [MaxLength(2000)]
    public string? Terms { get; set; }

    /// <summary>
    /// 备注
    /// </summary>
    [MaxLength(500)]
    public string? Remark { get; set; }

    /// <summary>
    /// 创建时间
    /// </summary>
    public DateTime CreateTime { get; set; }

    /// <summary>
    /// 更新时间
    /// </summary>
    public DateTime UpdateTime { get; set; }

    /// <summary>
    /// 签署时间
    /// </summary>
    public DateTime? SignTime { get; set; }

    /// <summary>
    /// 完成时间
    /// </summary>
    public DateTime? CompleteTime { get; set; }

    /// <summary>
    /// 取消时间
    /// </summary>
    public DateTime? CancelTime { get; set; }
} 

File: RealEstateSolution.VueAdmin/src/api/contract.ts
import request from './request'

// 合同类型枚举
export enum ContractType {
  Sale = 0,       // 买卖合同
  Rent = 1,       // 租赁合同
  Commission = 2  // 委托合同
}

// 合同状态枚举
export enum ContractStatus {
  Draft = 0,      // 草稿
  Pending = 1,    // 待签署
  Signed = 2,     // 已签署
  Completed = 3,  // 已完成
  Cancelled = 4   // 已取消
}

// 合同接口 - 统一用于增删改查
export interface Contract {
  id: number
  contractNumber: string
  title: string
  type: ContractType
  status: ContractStatus
  propertyId: number
  propertyTitle?: string
  propertyAddress?: string
  partyAId: number
  partyAName?: string
  partyAPhone?: string
  partyBId: number
  partyBName?: string
  partyBPhone?: string
  amount: number
  signDate?: string
  startDate?: string
  endDate?: string
  paymentMethod?: string
  terms?: string
  notes?: string
  content?: string
  createdAt: string
  updatedAt: string
}

// 合同查询接口
export interface ContractQuery {
  pageIndex?: number
  pageSize?: number
  keyword?: string
  type?: ContractType
  status?: ContractStatus
  startDate?: string
  endDate?: string
  partyAId?: number
  partyBId?: number
  propertyId?: number
}

// 合同模板接口
export interface ContractTemplate {
  id: number
  name: string
  description: string
  type: ContractType
  content: string
  isActive: boolean
  createdAt: string
  updatedAt: string
}

// 合同统计接口
export interface ContractStats {
  totalContracts: number
  draftContracts: number
  pendingContracts: number
  signedContracts: number
  completedContracts: number
  cancelledContracts: number
  totalAmount: number
  monthlyAmount: number
}

// 获取合同列表
export const getContracts = (params: ContractQuery) => {
  return request.get('/api/Contract/GetContracts', { params })
}

// 获取合同详情
export const getContract = (id: number) => {
  return request.get(`/api/Contract/GetContract/${id}`)
}

// 创建合同
export const createContract = (data: Contract) => {
  return request.post('/api/Contract/CreateContract', data)
}

// 更新合同
export const updateContract = (id: number, data: Contract) => {
  return request.put(`/api/Contract/UpdateContract/${id}`, data)
}

// 删除合同
export const deleteContract = (id: number) => {
  return request.delete(`/api/Contract/DeleteContract/${id}`)
}

// 更新合同状态
export const updateContractStatus = (id: number, status: ContractStatus) => {
  return request.post(`/api/Contract/UpdateContractStatus/${id}`, status)
}

// 获取合同模板列表
export const getContractTemplates = (type?: ContractType) => {
  return request.get('/api/Contract/GetContractTemplates', { params: { type } })
}

// 根据模板创建合同
export const createContractFromTemplate = (templateId: number, data: Contract) => {
  return request.post(`/api/Contract/CreateContractFromTemplate/${templateId}`, data)
}

// 生成合同编号
export const generateContractNumber = (type: ContractType) => {
  return request.get('/api/Contract/GenerateContractNumber', { params: { type } })
}

// 获取合同统计
export const getContractStats = () => {
  return request.get('/api/Contract/GetContractStats')
} 